test/e2e/singlecluster/e2e_test.go

```diff
@@ -141,13 +137,12 @@ var _ = ginkgo.Describe("Kueue", func() {
 				"instance-type": "on-demand",
 			})
 			wlLookupKey := types.NamespacedName{Name: workloadjob.GetWorkloadNameForJob(sampleJob.Name, sampleJob.UID), Namespace: ns.Name}
-			gomega.Eventually(func() bool {
-				if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
-					return false
-				}
-				return workload.HasQuotaReservation(createdWorkload) &&
-					apimeta.IsStatusConditionTrue(createdWorkload.Status.Conditions, kueue.WorkloadFinished)
-			}, util.LongTimeout, util.Interval).Should(gomega.BeTrue())
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).Should(gomega.Succeed())
+				g.Expect(workload.HasQuotaReservation(createdWorkload) &&
+					apimeta.IsStatusConditionTrue(createdWorkload.Status.Conditions, kueue.WorkloadFinished),
```

Comment:
To stay true to the spirit of gomega, we should probably have dedicated gomega matchers for conditions.

But that can be for a follow up.